Add Arabic locale

randa-11295 opened the pull request:
This branch adds Arabic language support to the project, including localisation and translations. RTL adjustments will be handled in a separate step if necessary

m-mohr commented on src/locales/ar/datepicker.js:
```diff
@@ -0,0 +1,54 @@
+// 1. Specify output format
```
You can remove the comments from the template file.

m-mohr commented on src/locales/ar/duration.js:
```diff
@@ -0,0 +1,33 @@
+// A. Either re-use phrases from iso-duration...
```
Same here, you can remove the comments from the template.

m-mohr commented on src/locales/ar/stac-fields.js:
```diff
@@ -0,0 +1,21 @@
+import I18N from '@radiantearth/stac-fields/I18N';
```
Do you use ISO formatting for date and time in Arabic? If not, the file should be removed.

randa-11295 replied:
Arabic does use ISO 8601 in technical and official contexts, but in daily life, people mostly use DD-MM-YYYY or, in some countries, Hijri dates (Islamic calendar - 15 Sha'ban 1447  ). If this file is overriding ISO formatting unnecessarily, I can remove it. Let me know what you think

m-mohr commented on src/locales/ar/texts.json:
```diff
@@ -0,0 +1,376 @@
+    "selectLabel": "اضغط على Enter للاختيار",
+    "selectedLabel": "تم الاختيار"
+  },
+  "open": "Open",
+  "pagination": {
+    "first": "« First",
+    "last": "Last »",
+    "next": "Next ›",
+    "previous": "‹ Previous"
+  },
+  "poweredBy": "Powered by {link}",
```
Translations missing. Untranslated phrases should be removed from the file until they are translated.

m-mohr commented on src/locales/ar/texts.json:
```diff
@@ -0,0 +1,376 @@
+  "poweredBy": "Powered by {link}",
+  "protocol": {
+    "azure": "Microsoft Azure",
+    "file": "file system (local)",
```
Translations missing. Untranslated phrases should be removed from the file until they are translated.

m-mohr reviewed (changes requested):
Thank you very much! This is an awesome addition to STAC Browser.

I've left some comments for your consideration.

What's your experience around RTL? Happy to collaborate on improvements that are needed for RTL support.

Also, I'd like to invite you to join Crowdin, where we manage the translations.
Which email adress shall I use for the invite?

randa-11295 commented:
Thank you for your feedback! I’ll review your comments and make the needed improvements.

I have good experience with RTL since I work with both Arabic and English. I'd love to help improve RTL support.

It would be great to join the Crowdin community! Please send the invite to 
randa.mohamed1295@gmail.com 
Looking forward to contributing!

randa-11295 commented:
Thank you for your feedback! I’ve reviewed your comments and made the necessary updates:
-Fixed translation inconsistencies.
-Updated fields.json with accurate Arabic translations.
-Removed unnecessary comments from datepicker and duration components.
Let me know if there are any specific areas that still need adjustments.

m-mohr commented:
Thank you very much, the PR looks very good now. I've sent you an invite to join the CrowdIn translation tool. If you did not receive it, please let me know and I'll resend it.

> I have good experience with RTL since I work with both Arabic and English. I'd love to help improve RTL support.

Wonderful. I meant to ask, what's your experience with RTL in STAC Browser? I think we need some adjustments there to cater better for Arabic and RTL. For example, set the `dir` property in HTML. This likely needs a new config option in the config.js locales.
I've opened a separate issue where we can discuss further steps: #574 I'd be happy to collaborate with you on this issue. As you are a web dev, feel free to start a PR if you want.
